Source/KSCrash/Recording/Tools/KSJSONCodec.c

```diff
@@ -1137,8 +1137,19 @@ int ksjsoncodec_i_decodeElement(const char** ptr,
                 return KSJSON_ERROR_INCOMPLETE;
             }
 
+            // our buffer is not necessarily NULL-terminated, so
+            // it would be undefined to call sscanf/sttod etc. directly.
+            // instead we create a temporary string.
             double value;
-            sscanf(start, "%lg", &value);
+            size_t len = (unsigned)(*ptr - start);
+            char * buf = malloc(len + 1);
```

Comment:
Maybe __svfscanf or something similar can be used here then.
Like here http://www.opensource.apple.com/source/Libc/Libc-167/stdio.subproj/sscanf.c 
But we know the length in this case and can omit strlen.